### PR TITLE
Add mobile menu to home page

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/) 🔎
- Asana tasks 🎟️ 
  - [Add mobile navigation menu on the home page](https://app.asana.com/0/1202022787106817/1202227071718194/f)
  - [Add grayed out products & beta badges to subnav menus](https://app.asana.com/0/1202097197789424/1202292475408431/f)

## 🗒️ What

- Adds `MobileMenuContainer`, which handles rendering the mobile menu correctly based on the open/closed state
- Adds `MobileMenuProvider` for managing mobile menu open/closed state and a `useMobileMenu` hook to expose the state and state updaters
  - The provider also handles no scroll on the body and closing on route changes
- Updates `CoreDevDotLayout` to render `MobileMenuProvider`
- Updates references to `sidebarIsOpen` and `setSidebarIsOpen` to use `useMobileMenu` instead
- Updates `SidebarNavDataContext` to leverage `useMobileMenu`
- Updates `SidebarSidecarLayout` to leverage `MobileMenuContainer` and `useMobileMenu`
- Fixes `cursor` setting in "disabled" `NavigationHeaderDropdownMenu` items
- Adds new `generateTopLevelSubNavItems` helper
- Updates `generateTopLevelSidebarNavData` to leverage new `generateTopLevelSubNavItems`
- Renames `generateTopLevelSidebarNavData`'s file to match the function name
- Updates `SidebarNavMenuItemBadge` to support multiple `type`s
- Updates `Sidebar` swingset page to show multiple badge types
- Updates `SidebarNavLinkItem` to:
  - support all product icons as leading icon
  - render a "disabled" link if no `href` is given
- Updates `BaseNewLayout` to render a mobile menu if the current page is the home page

## 📸 Design Screenshots

Figma link: https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/?node-id=9592%3A141879

CleanShot 2022-06-08 at 13 31 21@2x

## 🧪 Testing

We need to do full QA on all mobile menus & sidebars at desktop-up and desktop-down  on the following pages:

- [home](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/)
- [/vault](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault)
- [/vault/api-docs](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/api-docs)
- [/vault/docs](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/docs)
- [/vault/downloads](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/downloads)
- [/vault/tutorials](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/tutorials)
- [/vault/tutorials/getting-started](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/tutorials/getting-started)
- [/vault/tutorials/getting-started/getting-started-intro](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/vault/tutorials/getting-started/getting-started-intro)
- [/waypoint](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint)
- [/waypoint/commands](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/commands)
- [/waypoint/docs](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/docs)
- [/waypoint/plugins](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/plugins)
- [/waypoint/downloads](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/downloads)
- [/waypoint/tutorials/get-started-kubernetes](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/tutorials/get-started-kubernetes)
- [/waypoint/tutorials/get-started-kubernetes/get-started-intro](https://dev-portal-git-ambhome-page-mobile-menu-hashicorp.vercel.app/waypoint/tutorials/get-started-kubernetes/get-started-intro)

### CODE REVIEW

**Comment by BRKalow** [comment]
@ashleemboyer Seeing this on large viewports:
Screen Shot 2022-06-08 at 12 50 55 PM

**Comment by ashleemboyer** [comment]
> @ashleemboyer Seeing this on large viewports: Screen Shot 2022-06-08 at 12 50 55 PM

@BRKalow lol OOOF just pushed up a fix in https://github.com/hashicorp/dev-portal/pull/571/commits/403b3fe10ed3234b9b3953fbecfd4c48856587a3 🤣

File: src/layouts/base-new/index.tsx [hunk 1]
```diff
@@ -1,38 +1,66 @@
...
+                  <SidebarNavMenuItem item={{ heading: 'Main Menu' }} />
+                  {generateTopLevelSubNavItems().map(
+                    (item: $TSFixMe, index: number) => (
+                      // eslint-disable-next-line react/no-array-index-key
+                      <SidebarNavMenuItem item={item} key={index} />
+                    )
+                  )}
+                </ul>
+              </MobileMenuContainer>
+            )}
```
[the remark this answers is not in the document]
Comment (ashleemboyer) [reply src/layouts/base-new/index.tsx]: Made this update in https://github.com/hashicorp/dev-portal/pull/571/commits/6db2a34d701c7b71e87ea09152dc6e32ffa6850d. Great call!

**Comment by braydenlove** [comment]
I initially saw what Bryce pointed out above, just checked it out again and it looks really good. I will wait until the checks and preview build are done to give my final thumbs up! Nice work 🥇

**Review by zchsh — APPROVED** [review]
✅  I've only skimmed the code, but have done a pretty thorough QA of `/vault` pages and everything looks good to me 👍 Going to take a look at `/waypoint` after next meeting.

EDIT: certainly no reason to expect otherwise, but `/waypoint` looks great too 💯

**Review by BRKalow — APPROVED** [review]
looks good! 👍

**Pull request merged**